src/compile/data/optimizers.ts

```diff
@@ -78,3 +78,29 @@ export function removeUnusedSubtrees(node: DataFlowNode) {
   }
   return true;
 }
+
+/**
+ * Generates time unit nodes from selection filters and removes duplicate
+ * time unit nodes (as determined by the name of the output field)
+ * that may be generated due to selections projected over time units.
+ */
+export function cleanupSelectionTimeUnits(leaf: DataFlowNode) {
+  let fields = {};
+  return iterateFromLeaves((node: DataFlowNode) => {
+    if (node instanceof FilterNode) {
+      // Selection filters only add TimeUnitNodes on assembly.
```

Comment:
If we change 3. and parse selections before we parse selections, would it be possible to insert the timeunit node when we parse the (selection) filter?